### PR TITLE
[NFC] Avoid verifying module in release builds

### PR DESCRIPTION
**Pull request by AlexeySachkov**
This is achieved by guarding most (only covered LLVM IR -> SPIR-V passes)
`verifyModule` calls by corresponding command line options, which are
disabled/enabled in accordance with _SPIRVDBG define.

Made _SPIRVDBG define disabled in release builds by default

### CODE REVIEW

**Review by AlexeySotkin — COMMENTED** [review]
What about having _one_ command line option which enables or disables validation/verification of LLVM module after _every_ pass?

File: lib/SPIRV/libSPIRV/SPIRVDebug.h [hunk 1]
```diff
@@ -72,7 +84,43 @@ inline spv_ostream &spvdbgs() {
...
+
+// Output stream for SPIRV debug information.
+inline dev_null_stream &spvdbgs() {
+  static dev_null_stream Out;
+  return Out;
+}
+
 #endif
 
+#define VERIFY_REGULARIZATION_PASS(Module, PassName)                           \
```
Comment (svenvh) [inline lib/SPIRV/libSPIRV/SPIRVDebug.h:112]: Any reason why this has to be a macro?

File: lib/SPIRV/libSPIRV/SPIRVDebug.cpp [hunk 2]
```diff
@@ -45,3 +45,8 @@ bool SPIRV::SPIRVDbgEnable = false;
 SPIRV::SPIRVDbgErrorHandlingKinds SPIRV::SPIRVDbgError =
     SPIRVDbgErrorHandlingKinds::Exit;
 bool SPIRV::SPIRVDbgErrorMsgIncludesSourceInfo = true;
+
+llvm::cl::opt<bool> SPIRV::VerifyRegularizationPasses(
+    "spirv-verify-regularize-passes", llvm::cl::init(_SPIRVDBG),
+    llvm::cl::desc(
+        "Validate module after each pass in LLVM regularization phase"));
```
Comment (svenvh) [inline lib/SPIRV/libSPIRV/SPIRVDebug.cpp:52]:
```suggestion
        "Verify module after each pass in LLVM regularization phase"));
```

File: lib/SPIRV/libSPIRV/SPIRVDebug.h (hunk 1 above)
Comment (AlexeySachkov) [reply lib/SPIRV/libSPIRV/SPIRVDebug.h:112]: > Any reason why this has to be a macro?

No particular reason: I agree that macro is completely unnecessary, will refactor into a helper function instead

Comment (AlexeySachkov) [reply lib/SPIRV/libSPIRV/SPIRVDebug.h]: Replaced macro with a helper function in 33c039b949569438afeba506ef8dfa7c56ed2a17

File: lib/SPIRV/libSPIRV/SPIRVDebug.cpp (hunk 2 above)
Comment (AlexeySachkov) [reply lib/SPIRV/libSPIRV/SPIRVDebug.cpp]: Applied in 33c039b949569438afeba506ef8dfa7c56ed2a17

**Review by svenvh — APPROVED** [review]

**Pull request merged**